### ISSUE TITLE
Error when resuming training from mBART checkpoint

### ISSUE THREAD

**Issue by theamato** [MGheini/xattn-transfer-for-mt#3]
Hi,

I'm trying to run `xattn-transfer-for-mt/scripts/training/reinit-src-embeddings-and-finetune-parent-model-on-translation_src+xattn.sh` on my own dataset with the mBART checkpoint (instead of the FrEn- checkpoint), but when I do I get this error: 

[--encoder-ffn-embed-dim ENCODER_FFN_EMBED_DIM]
                [--encoder-layers ENCODER_LAYERS]
                [--encoder-attention-heads ENCODER_ATTENTION_HEADS]
                [--encoder-normalize-before] [--encoder-learned-pos]
                [--encoder-layerdrop ENCODER_LAYERDROP]
                [--encoder-layers-to-keep ENCODER_LAYERS_TO_KEEP]
                [--encoder-xformers-att-config ENCODER_XFORMERS_ATT_CONFIG]
                [--max-source-positions MAX_SOURCE_POSITIONS]
                [--decoder-embed-path DECODER_EMBED_PATH]
                [--decoder-embed-dim DECODER_EMBED_DIM]
                [--decoder-ffn-embed-dim DECODER_FFN_EMBED_DIM]
                [--decoder-layers DECODER_LAYERS]
                [--decoder-attention-heads DECODER_ATTENTION_HEADS]
                [--decoder-normalize-before] [--decoder-learned-pos]
                [--decoder-layerdrop DECODER_LAYERDROP]
                [--decoder-layers-to-keep DECODER_LAYERS_TO_KEEP]
                [--decoder-xformers-att-config DECODER_XFORMERS_ATT_CONFIG]
                [--decoder-output-dim DECODER_OUTPUT_DIM]
                [--max-target-positions MAX_TARGET_POSITIONS]
                [--share-decoder-input-output-embed] [--share-all-embeddings]
                [--no-token-positional-embeddings]
                [--adaptive-softmax-cutoff ADAPTIVE_SOFTMAX_CUTOFF]
                [--adaptive-softmax-dropout ADAPTIVE_SOFTMAX_DROPOUT]
                [--adaptive-softmax-factor ADAPTIVE_SOFTMAX_FACTOR]
                [--layernorm-embedding] [--tie-adaptive-weights]
                [--tie-adaptive-proj] [--no-scale-embedding]
                [--checkpoint-activations] [--offload-activations]
                [--no-cross-attention] [--cross-self-attention]
                [--quant-noise-pq QUANT_NOISE_PQ]
                [--quant-noise-pq-block-size QUANT_NOISE_PQ_BLOCK_SIZE]
                [--quant-noise-scalar QUANT_NOISE_SCALAR]
                [--min-params-to-wrap MIN_PARAMS_TO_WRAP] [--char-inputs]
                [--relu-dropout RELU_DROPOUT] [--base-layers BASE_LAYERS]
                [--base-sublayers BASE_SUBLAYERS]
                [--base-shuffle BASE_SHUFFLE] [--export]
                [--no-decoder-final-norm] [--source-lang SOURCE_LANG]
                [--target-lang TARGET_LANG] [--load-alignments]
                [--left-pad-source] [--left-pad-target]
                [--upsample-primary UPSAMPLE_PRIMARY] [--truncate-source]
                [--num-batch-buckets NUM_BATCH_BUCKETS] [--eval-bleu]
                [--eval-bleu-args EVAL_BLEU_ARGS]
                [--eval-bleu-detok EVAL_BLEU_DETOK]
                [--eval-bleu-detok-args EVAL_BLEU_DETOK_ARGS]
                [--eval-tokenized-bleu]
                [--eval-bleu-remove-bpe [EVAL_BLEU_REMOVE_BPE]]
                [--eval-bleu-print-samples]
                [--label-smoothing LABEL_SMOOTHING] [--report-accuracy]
                [--ignore-prefix-size IGNORE_PREFIX_SIZE]
                [--adam-betas ADAM_BETAS] [--adam-eps ADAM_EPS]
                [--weight-decay WEIGHT_DECAY] [--use-old-adam]
                [--fp16-adam-stats] [--warmup-updates WARMUP_UPDATES]
                [--force-anneal FORCE_ANNEAL]
                [--end-learning-rate END_LEARNING_RATE] [--power POWER]
                [--total-num-update TOTAL_NUM_UPDATE] [--pad PAD] [--eos EOS]
                [--unk UNK]
                data
train.py: error: unrecognized arguments: --min-lr -1 --load-model-but-src-embeddings-and-freeze-tgt-embeddings-from ../FrEn/checkpoints/checkpoint_best.pt --only-finetune-cross-attn

Why does it not recognize some of the arguments? And why does it load the FrEn checkpoint when I set the path to the desired model in the script? 

Any input would be much appreciated.

**Comment by MGheini** [comment]
Hi,

Can you copy and paste the script as you're trying to run here? Are you using the modified `fairseq` as included in this repo?
Additionally, I'd recommend using provided mBART-specific scripts for fine-tuning mBART: `finetune-mbart-on-transaltion_embed+xattn.sh`.

**Comment by ghost** [comment]
This is the whole ommand I'm trying to run (I did as you said and swapped to the `finetune-mbart-on-transaltion_embed+xattn.sh-` script:

`!bash 'content/xattn-transfer-for-mt/scripts/training/finetune-mbart-on-transaltion_embed+xattn.sh' \
    /path/to/bpe/data som en mbart.cc25/model.pt` 

I did set the fairseq environment variable as `%env` `FAIRSEQ=/content/xattn-transfer-for-mt/fairseq-modified` (I'm using colab). 

This is the finetune-mbart-on-transaltion_embed+xattn.sh- script (although I did not change anything here). 

`DIR=$1
cd $DIR

SRC=$2
TGT=$3

MBART=$4

langs=ar_AR,cs_CZ,de_DE,en_XX,es_XX,et_EE,fi_FI,fr_XX,gu_IN,hi_IN,it_IT,ja_XX,kk_KZ,ko_KR,lt_LT,lv_LV,my_MM,ne_NP,nl_XX,ro_RO,ru_RU,si_LK,tr_TR,vi_VN,zh_CN

python $FAIRSEQ/fairseq_cli/train.py data-bin \
	--langs $langs \
	--source-lang $SRC --target-lang $TGT \
	--log-format simple \
	--log-interval 20 \
	--seed 222 \
	--criterion label_smoothed_cross_entropy \
	--label-smoothing 0.2 \
	--optimizer adam \
	--adam-eps 1e-06 \
	--adam-betas "(0.9, 0.98)" \
	--weight-decay 0.0 \
	--lr-scheduler polynomial_decay \
	--task translation_from_pretrained_bart \
	--eval-bleu --eval-bleu-detok moses \
	--num-workers 8 \
	--max-tokens 512 \
	--validate-interval 1 \
	--arch mbart_large \
	--max-update 150000 \
	--update-freq 8 \
	--lr 3e-05 \
	--min-lr -1 \
	--restore-file checkpoint_last.pt \
	--save-interval 1 \
	--save-interval-updates 500 \
	--keep-interval-updates 1 \
	--no-epoch-checkpoints \
	--warmup-updates 2500 \
	--dropout 0.3 \
	--attention-dropout 0.1 \
	--relu-dropout 0.0 \
	--layernorm-embedding \
	--encoder-learned-pos \
	--decoder-learned-pos \
	--encoder-normalize-before \
	--decoder-normalize-before \
	--skip-invalid-size-inputs-valid-test \
	--share-all-embeddings \
	--finetune-from-mbart-at $MBART \
	--only-finetune-cross-attn \
	--patience 25`

**Comment by ghost** [comment]
Actually, turns out I wasn't using fairseq modified after all. I had to add `sys.path.insert(0, '/content/xattn-transfer-for-mt/fairseq-modified') ` to train.py, and it's running fine now. Thank you!
